daisy/workflow.go

```diff
@@ -894,7 +887,22 @@ func (w *Workflow) IterateWorkflowSteps(cb func(step *Step)) {
 // CancelWithReason cancels workflow with a specific reason. The specific reason replaces "is canceled" in the default error message.
 func (w *Workflow) CancelWithReason(reason string) {
 	w.cancelReason = reason
-	close(w.Cancel)
+	w.CancelWorkflow()
+}
+
+// CancelWorkflow cancels the workflow. Safe to call multiple times.
+// Prefer this to closing the w.Cancel channel,
+// which will panic if it has already been closed.
+func (w *Workflow) CancelWorkflow() {
+	w.cleanupHooksMx.Lock()
```

Comment:
should be inCanceledMx per my understanding

Reply:
shoot, that was a typo! Thanks; fixed.